lib/rspec/core/filter_manager.rb

```diff
@@ -99,13 +99,17 @@ def prune_conditionally_filtered_examples(examples)
       # defined in the same file as the location filters. Excluded specs in
       # other files should still be excluded.
       def file_scoped_include?(ex_metadata, ids, locations)
-        no_location_filters = locations[ex_metadata[:absolute_file_path]].empty?
-        no_id_filters = ids[ex_metadata[:rerun_file_path]].empty?
-
-        return yield if no_location_filters && no_id_filters
-
-        MetadataFilter.filter_applies?(:ids, ids, ex_metadata) ||
-        MetadataFilter.filter_applies?(:locations, locations, ex_metadata)
+        has_id_filters = ids[ex_metadata[:rerun_file_path]].any?
+        has_location_filters = locations[
+          File.expand_path(ex_metadata[:rerun_file_path])
+        ].any?
+
+        if has_location_filters || has_id_filters
```

Comment:
I think I'd prefer that this was expressed as a guard condition like before, e.g.
`return yield unless has_location_filters || has_id_filters`